### ISSUE TITLE
Rule to prevent awaiting task that can just be returned.

### ISSUE THREAD

**Issue by mjr4077au** [meziantou/Meziantou.Analyzer#1280]
**Is your feature request related to a problem? Please describe.**

Consider the below method:
```c#
internal static async Task CloseNotifyIconAsync()
{
    await InvokeDialogActionAsync(() =>
    {
        using (notifyIcon)
        {
            lastBalloonTip = null;
            notifyIcon = null;
        }
    }).ConfigureAwait(false);
}
```
This could easily become:
```c#
internal static Task CloseNotifyIconAsync()
{
    return InvokeDialogActionAsync(() =>
    {
        using (notifyIcon)
        {
            lastBalloonTip = null;
            notifyIcon = null;
        }
    });
}
```

**Describe the solution you'd like**
I'd like a rule to enforce this if possible. Roslynator has RCS1174, but it's only wired up for tasks that return an object.

**Describe alternatives you've considered**
Manually auditing my code for such setups.

**Comment by meziantou** [comment]
The new rules are opt-in. Also, there are lots of cases, so I expect multiple false-positive. Create new issues to report them.
